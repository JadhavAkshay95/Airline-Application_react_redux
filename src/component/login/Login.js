import React, { useState } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import { connect } from 'react-redux';
import * as LoginAction from './../../redux/actions/loginAction';
import './login.scss';
import LoginSideInfo from './LoginSideInfo';
import LoginModal from './LoginModal';
import PropTypes from 'prop-types';

export const Login = ({ history, setCurrentUser, getCurrentUser }) => {
  const [show, setShow] = useState(false);

  const handleClose = () => {
    setShow(false);
  };

  const openLoginModal = () => {
    setShow(true);
  };

  const handleFbLogin = async response => {
    await setCurrentUser({
      role: 'staff',
      username: response.email
    });
    await getCurrentUser();
    history.push('/staff-home');
    handleClose();
  };

  const handleGoogleLogin = async response => {
    await setCurrentUser({
      role: 'admin',
      username: response.username
    });
    await getCurrentUser();

    history.push('/admin/flight-list');
    handleClose();
  };

  return (
    <Container fluid="true" className="login-container">
      <Row>
        <Col>
          <div className="login-logo">
            <img
              alt={require('../../images/aa.png')}
              src={require('../../images/aa.png')}
            />
          </div>
        </Col>
        <LoginModal
          show={show}
          handleClose={handleClose}
          handleGoogleLogin={handleGoogleLogin}
          handleFbLogin={handleFbLogin}
        />
        <LoginSideInfo showLoginModal={openLoginModal} />
      </Row>
    </Container>
  );
};

export const mapStateToProps = state => {
  return {
    currentUser: Array.isArray(state.loginReducer) ? null : state.loginReducer
  };
};

const mapDispatchToProps = {
  setCurrentUser: LoginAction.setCurrentUser,
  getCurrentUser: LoginAction.getCurrentUser
};
export default connect(mapStateToProps, mapDispatchToProps)(Login);
Login.propTypes = {
  history: PropTypes.object,
  setCurrentUser: PropTypes.func,
  getCurrentUser: PropTypes.func
};
