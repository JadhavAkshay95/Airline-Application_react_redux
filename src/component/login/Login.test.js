import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

const renderLogin = args => {
  const defaultProps = {
    history: {},
    loginInfo: {},
    getLoginInfo: () => {},
    setCurrentUser: () => {},
    currentUser: {},
    getCurrentUser: () => {}
  };
  const props = { ...defaultProps, ...args };
  return shallow(<Login {...props} />);
};

describe('Render login component', () => {
  it('Should render login component', () => {
    const wrapper = renderLogin();
    expect(wrapper).toMatchSnapshot();
  });

  it('should close modal', () => {
    const wrapper = renderLogin();
    const parentProps = wrapper.find('LoginModal').props();
    parentProps.handleClose();
    expect(parentProps.show).toBeFalsy();
  });

  it('should open modal', () => {
    const wrapper = renderLogin();
    const parentProps = wrapper.find('LoginSideInfo').props();
    parentProps.showLoginModal();
  });
});
