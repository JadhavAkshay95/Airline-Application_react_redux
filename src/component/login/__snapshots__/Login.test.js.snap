// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render login component Should render login component 1`] = `ShallowWrapper {}`;
